### PR TITLE
Credo linting

### PR DESCRIPTION
**Pull request by bblaszkow06**
Circle now uses 2 parallel jobs, one builds in dev env, the other in test

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -11,8 +11,29 @@ jobs:
...
+      - run: mix compile
+      - run: mix credo
+      - run: mix docs && ! mix docs 2>&1 | grep -q "warning:"
+
+workflows:
+  version: 2
+  build:
+    jobs:
+      - test
+      - lint
```
Comment (bblaszkow06) [inline .circleci/config.yml:39]: The difference is ~ 1min:
https://app.circleci.com/pipelines/github/membraneframework/membrane_core/735/workflows/7f53c7d5-cffc-4384-b5d8-7d11ce255c81
https://app.circleci.com/pipelines/github/membraneframework/membrane_core/734/workflows/9f9c0dd1-75d0-4fd7-9092-44de6f873342

**Comment by bblaszkow06** [comment]
The GitHub checks are stuck because of settings: task `build` is required, but only `test` and `lint` are now run. After approval these settings can be changed

File: lib/membrane/time.ex [hunk 2]
```diff
@@ -250,12 +250,14 @@ defmodule Membrane.Time do
     Returns one #{unit.singular} in `#{inspect(__MODULE__)}` units.
     """
     @spec unquote(unit.singular)() :: t
+    # credo:disable-for-next-line Credo.Check.Readability.Specs
```
Comment (mat-hek) [inline lib/membrane/time.ex:253]: isn't the spec present?

File: test/membrane/element_test.exs [hunk 3]
```diff
@@ -14,8 +14,10 @@ defmodule Membrane.ElementTest do
 
     def_options target: [type: :pid]
 
+    # credo:disable-for-next-line Credo.Check.Readability.Specs
     def assert_callback_called(name), do: assert_receive({:callback_called, ^name})
```
Comment (mat-hek) [inline test/membrane/element_test.exs:18]: does this need to be public?

File: lib/membrane/time.ex (hunk 2 above)
Comment (bblaszkow06) [reply lib/membrane/time.ex:253]: Credo does see that :man_shrugging:

File: test/membrane/element_test.exs (hunk 3 above)
Comment (bblaszkow06) [reply test/membrane/element_test.exs:18]: If placed in this module - yes because it's used in ElementTest

Comment (mat-hek) [reply test/membrane/element_test.exs:18]: is there any reason why it's placed there?

Comment (bblaszkow06) [reply test/membrane/element_test.exs:18]: Yep - cuz it's specific for this element (matches atom sent in callbacks) and you can call it like this: `TestFilter.assert_callback_called(:handle_end_of_stream)`

Comment (mat-hek) [reply test/membrane/element_test.exs:18]: So maybe just add specs for that

**Pull request merged**